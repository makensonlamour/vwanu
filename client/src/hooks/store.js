import { configureStore } from "@reduxjs/toolkit";
import authReducer from "../features/auth/authSlice";
import alertReducer from "../features/alert/alertSlice";
import userReducer from "../features/user/userSlice";
import { apiSlice } from "../features/api/apiSlice";

export const store = configureStore({
  reducer: {
    user: userReducer,
    alert: alertReducer,
    auth: authReducer,
    [apiSlice.reducerPath]: apiSlice.reducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(apiSlice.middleware),
});
