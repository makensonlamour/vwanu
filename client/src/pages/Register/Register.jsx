import React from "react";
import { Link } from "react-router-dom";
import routesPath from "../../routesPath";

// Core components
import FormRegister from "../../features/auth/Register/FormRegister";
import logo from "../../assets/images/Vwanu_logo.svg";
import Shape_up from "../../assets/images/Shape_up.svg";
import Home_pic from "../../assets/images/Home_pic.jpg";
import { BsFacebook, BsTwitter } from "react-icons/bs";
import { FaGooglePlus } from "react-icons/fa";

const RegisterScreen = () => {
  return (
    <>
      <div className="mx-0 md:mx-0 lg:h-screen ">
        <div className="grid grid-cols-1 lg:mb:0 lg:grid-cols-2">
          <div
            style={{
              backgroundImage: `url(${Home_pic})`,
              backgroundPosition: "center",
              backgroundSize: "cover",
              backgroundRepeat: "no-repeat",
            }}
            className="hidden lg:inline lg:relative lg:justify-center"
          >
            <div className="bg-gradient-to-tr from-g-one/[0.78] to-g-two/[0.78] h-96 z-0 lg:h-screen">
              <img src={Shape_up} alt="_shape_up" className="hidden lg:block w-[22%] lg:absolute right-0 z-30 -mt-8 object-fit" />
              <p className="text-center pt-14 text-base-100 text-2xl py-5">
                <img className="w-[30%] m-auto" src={logo} alt="logo_vwanu" />
              </p>
              <p className="text-5xl text-v-yellow-dark font-bold py-10 mt-10 text-center align-middle">Welcome</p>
              <p className="hidden lg:block text-base-100 text-xl font-semibold text-center pb-2 align-middle">Together We are Stronger</p>
            </div>
            <img
              src={Shape_up}
              alt="_shape_down"
              className="hidden lg:block w-[23%] lg:absolute left-0 z-30 bottom-0 object-fit rotate-180"
            />
          </div>

          <div className="inline lg:hidden">
            <div className="bg-gradient-to-tr from-g-one/[0.78] to-g-two/[0.78] h-96 z-0 lg:h-screen">
              <img src={Shape_up} alt="_shape_up" className="hidden lg:block w-[22%] lg:absolute right-0 z-30 -mt-8 object-fit" />
              <p className="text-center pt-14 text-base-100 text-2xl py-5">
                <img className="w-[30%] m-auto" src={logo} alt="logo_vwanu" />
              </p>
              <p className="text-5xl text-v-yellow-dark font-bold py-10 mt-10 text-center">Welcome</p>
              <p className="hidden lg:block text-base-100 text-xl font-semibold text-center pb-2">Together We are Stronger</p>
            </div>
            <img
              src={Shape_up}
              alt="_shape_down"
              className="hidden lg:block w-[23%] lg:absolute left-0 z-30 bottom-0 object-fit rotate-180"
            />
          </div>

          <div className="place-items-center bg-base-100 shadow-t-2xl rounded-t-[30px] lg:rounded-none px-4 md:px-8 -mt-28 z-10 md:mx-36 lg:mt-0 lg:mx-0 lg:h-screen">
            <div className="lg:place-content-center lg:mt-6 lg:mb-3">
              <p className="text-center mt-4 lg:mt-0">
                <span className="text-md text-primary font-semibold md:text-xl">Already have an account ?</span>{" "}
                <Link
                  data-testid="loginBtn"
                  to={routesPath.LOGIN}
                  className="btn btn-primary btn-sm rounded-full normal-case hover:btn-secondary hover:text-base-100 md:px-8 ml-1 text-base-100"
                >
                  Sign In
                </Link>
              </p>
            </div>
            <div className="hidden lg:block mb-8 lg:mb-4 text-center">
              <span className="text-blue-600 text-center inline text-md md:text-lg">
                {` Or sign in with `}
                <Link className="text-blue-500 ml-2" to={"#"}>
                  <BsFacebook className="text-2xl inline mx-1" />
                </Link>
                <Link className="text-sky-400" to={"#"}>
                  <BsTwitter className="text-2xl inline mx-1" />
                </Link>
                <Link className="text-red-600" to={"#"}>
                  <FaGooglePlus className="text-2xl inline mx-1" />
                </Link>
              </span>
            </div>
            <FormRegister />
            <div className="lg:hidden mb-8 text-center">
              <span className="text-blue-600 text-center inline text-md md:text-lg">
                {` Or sign in with `}
                <Link className="text-blue-500 ml-2" to={"#"}>
                  <BsFacebook className="text-2xl inline mx-1" />
                </Link>
                <Link className="text-sky-400" to={"#"}>
                  <BsTwitter className="text-2xl inline mx-1" />
                </Link>
                <Link className="text-red-600" to={"#"}>
                  <FaGooglePlus className="text-2xl inline mx-1" />
                </Link>
              </span>
            </div>
          </div>
        </div>
        {/*}
        <div className="bg-gray-100 shadow-lg rounded-t-lg lg:mx-6 px-6 py-6 mt-10">
          <span className="text-secondary font-semibold">Language:</span>
        </div>
          {*/}
      </div>
    </>
  );
};

export default RegisterScreen;
