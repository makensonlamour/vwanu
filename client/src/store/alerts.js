import { createSlice } from "@reduxjs/toolkit";
const initialState = {
  list: [],
};
const Alert = createSlice({
  name: "alerts",
  initialState,
  reducers: {
    setAlert: (state, action) => {
      state.list.push(action.payload);
    },
    removeAlert: (state, action) => {
      const firstIndex = state.list.findIndex((alert) => alert.id === action.payload);

      state.list.splice(firstIndex, 1);
    },
  },
});

export default Alert.reducer;
export const { removeAlert, setAlert } = Alert.actions;
export const getAlerts = (state) => state.alerts;
