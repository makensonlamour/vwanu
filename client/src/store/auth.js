import * as action from "./api";
import env from "../config";
import { createSlice } from "@reduxjs/toolkit";
import jwtDecode from "jwt-decode";
import { saveToken, deleteToken } from "../middleware/api";

const url = env.endpoints;
//const { EXPECTED_HEADER } = env;

const initialState = {
  loading: false,
  data: null,
  lastFetch: null,
  token: null,
  error: null,
};

export const Auth = createSlice({
  name: "auth",
  initialState,
  reducers: {
    loginRequested: (state) => {
      state.loading = true;
    },
    loginSucceed: (state, action) => {
      state.loading = false;
      state.token = action.payload.data.token;
      saveToken(action.payload.data.token);
      //state.data = action.payload.data.user;
      state.data = jwtDecode(action.payload.data.token);
      state.lastFetch = Date.now;
    },

    LoginFailed: (state, action) => {
      state.loading = false;
      state.token = null;
      state.error = action.payload;
      state.data = null;
      deleteToken();
    },
    setUser: (state, action) => {
      state.loading = false;
      state.token = action.payload;
      saveToken(action.payload);
      state.data = jwtDecode(action.payload);
    },
    LogOut: (state) => {
      state.loading = false;
      state.token = null;
      state.data = null;
      state.error = null;
      deleteToken();
      window.location.replace("/login");
    },
  },
});

export const Login = (credentials) => (dispatch, getState) => {
  const { data, loading } = getState();
  if (data != null || loading) return;
  dispatch(
    action.apiCallBegan({
      url: url.LOGIN,
      data: credentials,
      method: "POST",
      onSuccess: Auth.actions.loginSucceed.type,
      onStart: Auth.actions.loginRequested.type,
      onError: Auth.actions.LoginFailed.type,
    })
  );
};

export const createUser = (newUserData) => (dispatch) => {
  dispatch(
    action.apiCallBegan({
      url: url.REGISTER,
      data: newUserData,
      method: "POST",
      onSuccess: Auth.actions.loginSucceed.type,
      onStart: Auth.actions.loginRequested.type,
      onError: Auth.actions.LoginFailed.type,
    })
  );
};

export const forgotPassword = (newUserData) => (dispatch) => {
  dispatch(
    action.apiCallBegan({
      url: url.FORGOT_PASSWORD,
      data: newUserData,
      method: "POST",
      onSuccess: Auth.actions.loginSucceed.type,
      onStart: Auth.actions.loginRequested.type,
      onError: Auth.actions.LoginFailed.type,
    })
  );
};

export const resetPassword = (newUserData) => (dispatch) => {
  dispatch(
    action.apiCallBegan({
      url: url.RESET_PASSWORD,
      data: newUserData,
      method: "POST",
      onSuccess: Auth.actions.loginSucceed.type,
      onStart: Auth.actions.loginRequested.type,
      onError: Auth.actions.LoginFailed.type,
    })
  );
};

export const { setUser } = Auth.actions;
export const logout = () => (dispatch) => dispatch(Auth.actions.LogOut());
export const getCurrentUser = (state) => state.authentication;
export const logged = Auth.actions.loginSucceed.type;
export default Auth.reducer;
