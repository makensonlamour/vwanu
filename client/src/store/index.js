import { configureStore } from "@reduxjs/toolkit";
import thunkMiddleware from "redux-thunk";
import reducer from "./reducer";
import api from "../middleware/api";
import alerts from "../middleware/alerts";

export const store = configureStore({
  reducer,
  middleware: [thunkMiddleware, api, alerts],
});
