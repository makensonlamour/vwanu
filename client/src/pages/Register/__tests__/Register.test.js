import React from "react";
import { render as rtlRender, screen, cleanup, fireEvent, waitFor } from "@testing-library/react";
import { MemoryRouter as Router } from "react-router-dom";
import { act as testAct, create } from "react-test-renderer";
import { act as domAct } from "react-dom/test-utils";
import { store } from "../../../store";
import Register from "../Register";
import { Provider } from "react-redux";

afterEach(cleanup);

const render = (component) => rtlRender(<Provider store={store}>{component}</Provider>);

//Testing Register form
//Test Dom Elements
describe("Register Test Dom Elements", () => {
  it("match snapshot", () => {
    let tree;
    domAct(() => {
      testAct(() => {
        tree = create(
          <Provider store={store}>
            <Router>
              <Register />
            </Router>
          </Provider>
        );
      });
    });
    expect(tree).toMatchSnapshot();
  });
});

it("should have a Register form", () => {
  render(
    <Router>
      <Register />
    </Router>
  );

  const registerForm = screen.getByRole("form");
  expect(registerForm).toBeInTheDocument();
});

it("should have an email input", () => {
  render(
    <Router>
      <Register />
    </Router>
  );

  const emailInput = screen.getByPlaceholderText("Email");
  expect(emailInput).toBeInTheDocument();
});

it("have a password input", () => {
  render(
    <Router>
      <Register />
    </Router>
  );

  const passwordInput = screen.getByPlaceholderText("Password");
  expect(passwordInput).toBeInTheDocument();
});

it("have a checkbox input", () => {
  render(
    <Router>
      <Register />
    </Router>
  );

  const termCheckbox = screen.getAllByRole("checkbox")[0];
  expect(termCheckbox).toBeInTheDocument();
});

it("should have a Register button", () => {
  render(
    <Router>
      <Register />
    </Router>
  );

  const submitButton = screen.getByRole("button", { name: "Register" });
  expect(submitButton).toBeInTheDocument();
});

it("the submit button should have the text Register", () => {
  render(
    <Router>
      <Register />
    </Router>
  );

  const submitButton = screen.getByRole("button", { name: "Register" });
  expect(submitButton).toHaveTextContent("Register");
});

it("should have a register link", () => {
  render(
    <Router>
      <Register />
    </Router>
  );

  const registerLink = screen.getByTestId("loginBtn");
  expect(registerLink).toHaveTextContent("Login");
});

//Test require fields
describe("Test require fields", () => {
  it("the email input should be a required fields", () => {
    render(
      <Router>
        <Register />
      </Router>
    );

    const emailInput = screen.getByPlaceholderText("Email");
    expect(emailInput).toBeInTheDocument();
    expect(emailInput).toHaveAttribute("required");
  });

  it("the password input should be a required fields", () => {
    render(
      <Router>
        <Register />
      </Router>
    );

    const passwordInput = screen.getByPlaceholderText("Password");
    expect(passwordInput).toBeInTheDocument();
    expect(passwordInput).toHaveAttribute("required");
  });

  it("on initial render, the submit button is enabled", () => {
    render(
      <Router>
        <Register />
      </Router>
    );

    const submitButton = screen.getByRole("button", { name: "Register" });
    expect(submitButton).not.toBeDisabled();
  });
});

//test register functionality
describe("Test Register Functionality", () => {
  it(" should show error message on password, email and Term of Use when the form is submitted with empty fields", async () => {
    domAct(() => {
      render(
        <Router>
          <Register />
        </Router>
      );
    });

    const emailInput = screen.getByPlaceholderText("Email");
    const passwordInput = screen.getByPlaceholderText("Password");
    const termCheckbox = screen.getAllByRole("checkbox")[0];
    const submitButton = screen.getByRole("button", { name: "Register" });

    fireEvent.change(emailInput, { target: { value: "" } });
    fireEvent.change(passwordInput, { target: { value: "" } });
    fireEvent.change(termCheckbox, { target: { value: false } });

    submitButton.click();
    await waitFor(() => expect(screen.queryByTestId("email-error-message")).toHaveTextContent("Email is a required field"));
    await waitFor(() => expect(screen.queryByTestId("password-error-message")).toHaveTextContent("Password is a required field"));
    await waitFor(() =>
      expect(screen.queryByTestId("termOfUse-error-message")).toHaveTextContent("You must accept the terms of use and the policy privacy")
    );
  });

  it(" should show error message email when the form is submitted with empty email", async () => {
    domAct(() => {
      render(
        <Router>
          <Register />
        </Router>
      );
    });

    const emailInput = screen.getByPlaceholderText("Email");
    const passwordInput = screen.getByPlaceholderText("Password");
    const termCheckbox = screen.getAllByRole("checkbox")[0];
    const submitButton = screen.getByRole("button", { name: "Register" });

    fireEvent.change(emailInput, { target: { value: "" } });
    fireEvent.change(passwordInput, { target: { value: "Digicel.1" } });
    termCheckbox.click();

    submitButton.click();
    await waitFor(() => expect(screen.queryByTestId("email-error-message")).toHaveTextContent("Email is a required field"));
  });

  it(" should show error message password when the form is submitted with empty password", async () => {
    domAct(() => {
      render(
        <Router>
          <Register />
        </Router>
      );
    });

    const emailInput = screen.getByPlaceholderText("Email");
    const passwordInput = screen.getByPlaceholderText("Password");
    const termCheckbox = screen.getAllByRole("checkbox")[0];
    const submitButton = screen.getByRole("button", { name: "Register" });

    fireEvent.change(emailInput, { target: { value: "test@test.com" } });
    fireEvent.change(passwordInput, { target: { value: "" } });
    termCheckbox.click();

    submitButton.click();
    await waitFor(() => expect(screen.queryByTestId("password-error-message")).toHaveTextContent("Password is a required field"));
  });

  it(" should show error message Checkbox when the form is submitted when the checkbox is not check", async () => {
    domAct(() => {
      render(
        <Router>
          <Register />
        </Router>
      );
    });

    const emailInput = screen.getByPlaceholderText("Email");
    const passwordInput = screen.getByPlaceholderText("Password");
    const submitButton = screen.getByRole("button", { name: "Register" });

    fireEvent.change(emailInput, { target: { value: "test@test.com" } });
    fireEvent.change(passwordInput, { target: { value: "Digicel.1" } });

    submitButton.click();
    await waitFor(() =>
      expect(screen.queryByTestId("termOfUse-error-message")).toHaveTextContent("You must accept the terms of use and the policy privacy")
    );
  });

  it(" should show error message email if email missing @", async () => {
    domAct(() => {
      render(
        <Router>
          <Register />
        </Router>
      );
    });

    const emailInput = screen.getByPlaceholderText("Email");
    const passwordInput = screen.getByPlaceholderText("Password");
    const termCheckbox = screen.getAllByRole("checkbox")[0];
    const submitButton = screen.getByRole("button", { name: "Register" });

    fireEvent.change(emailInput, { target: { value: "testtest.com" } });
    fireEvent.change(passwordInput, { target: { value: "Digicel.1" } });
    termCheckbox.click();

    submitButton.click();
    await waitFor(() => expect(screen.queryByTestId("email-error-message")).toHaveTextContent("Email must be a valid email"));
  });

  it(" should show error message email if email missing .", async () => {
    domAct(() => {
      render(
        <Router>
          <Register />
        </Router>
      );
    });

    const emailInput = screen.getByPlaceholderText("Email");
    const passwordInput = screen.getByPlaceholderText("Password");
    const termCheckbox = screen.getAllByRole("checkbox")[0];
    const submitButton = screen.getByRole("button", { name: "Register" });

    fireEvent.change(emailInput, { target: { value: "test@testcom" } });
    fireEvent.change(passwordInput, { target: { value: "" } });
    termCheckbox.click();

    submitButton.click();
    await waitFor(() => expect(screen.queryByTestId("email-error-message")).toHaveTextContent("Email must be a valid email"));
  });

  it(" should show error message password if password is least than 8 characters .", async () => {
    domAct(() => {
      render(
        <Router>
          <Register />
        </Router>
      );
    });

    const emailInput = screen.getByPlaceholderText("Email");
    const passwordInput = screen.getByPlaceholderText("Password");
    const termCheckbox = screen.getAllByRole("checkbox")[0];
    const submitButton = screen.getByRole("button", { name: "Register" });

    fireEvent.change(emailInput, { target: { value: "test@test.com" } });
    fireEvent.change(passwordInput, { target: { value: "Digi" } });
    termCheckbox.click();

    submitButton.click();
    await waitFor(() => expect(screen.queryByTestId("password-error-message")).toHaveTextContent("Password must be at least 8 characters"));
  });

  it(" shouldn't display error", async () => {
    domAct(() => {
      render(
        <Router>
          <Register />
        </Router>
      );
    });

    const emailInput = screen.getByPlaceholderText("Email");
    const passwordInput = screen.getByPlaceholderText("Password");
    const termCheckbox = screen.getAllByRole("checkbox")[0];
    const submitButton = screen.getByRole("button", { name: "Register" });

    fireEvent.change(emailInput, { target: { value: "test@test.com" } });
    fireEvent.change(passwordInput, { target: { value: "Digicel.1" } });
    termCheckbox.click();

    submitButton.click();
    await waitFor(() => expect(screen.queryByTestId("password-error-message")).toBeNull());
    await waitFor(() => expect(screen.queryByTestId("email-error-message")).toBeNull());
    await waitFor(() => expect(screen.queryByTestId("termOfUse-error-message")).toBeNull());
  });
});
