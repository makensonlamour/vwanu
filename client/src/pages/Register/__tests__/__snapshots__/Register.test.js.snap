// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Test Dom Elements match snapshot 1`] = `
<div
  className="mx-0 md:mx-0 lg:h-screen "
>
  <div
    className="grid grid-cols-1 lg:mb:0 lg:grid-cols-2"
  >
    <div
      className="hidden lg:inline lg:relative lg:justify-center"
      style={
        Object {
          "backgroundImage": "url(Home_pic.jpg)",
          "backgroundPosition": "center",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
        }
      }
    >
      <div
        className="bg-gradient-to-tr from-g-one/[0.78] to-g-two/[0.78] h-96 z-0 lg:h-screen"
      >
        <img
          alt="_shape_up"
          className="hidden lg:block w-[22%] lg:absolute right-0 z-30 -mt-8 object-fit"
          src="Shape_up.svg"
        />
        <p
          className="text-center pt-14 text-base-100 text-2xl py-5"
        >
          <img
            alt="logo_vwanu"
            className="w-[30%] m-auto"
            src="Vwanu_logo.svg"
          />
        </p>
        <p
          className="text-5xl text-v-yellow-dark font-bold py-10 mt-10 text-center align-middle"
        >
          Welcome
        </p>
        <p
          className="hidden lg:block text-base-100 text-xl font-semibold text-center pb-2 align-middle"
        >
          Together We are Stronger
        </p>
      </div>
      <img
        alt="_shape_down"
        className="hidden lg:block w-[23%] lg:absolute left-0 z-30 bottom-0 object-fit rotate-180"
        src="Shape_up.svg"
      />
    </div>
    <div
      className="inline lg:hidden"
    >
      <div
        className="bg-gradient-to-tr from-g-one/[0.78] to-g-two/[0.78] h-96 z-0 lg:h-screen"
      >
        <img
          alt="_shape_up"
          className="hidden lg:block w-[22%] lg:absolute right-0 z-30 -mt-8 object-fit"
          src="Shape_up.svg"
        />
        <p
          className="text-center pt-14 text-base-100 text-2xl py-5"
        >
          <img
            alt="logo_vwanu"
            className="w-[30%] m-auto"
            src="Vwanu_logo.svg"
          />
        </p>
        <p
          className="text-5xl text-v-yellow-dark font-bold py-10 mt-10 text-center"
        >
          Welcome
        </p>
        <p
          className="hidden lg:block text-base-100 text-xl font-semibold text-center pb-2"
        >
          Together We are Stronger
        </p>
      </div>
      <img
        alt="_shape_down"
        className="hidden lg:block w-[23%] lg:absolute left-0 z-30 bottom-0 object-fit rotate-180"
        src="Shape_up.svg"
      />
    </div>
    <div
      className="place-items-center bg-base-100 shadow-t-2xl rounded-t-[30px] lg:rounded-none px-4 md:px-8 -mt-28 z-10 md:mx-36 lg:mt-0 lg:mx-0 lg:h-screen"
    >
      <div
        className="lg:place-content-center lg:mt-6 lg:mb-3"
      >
        <p
          className="text-center mt-4 lg:mt-0"
        >
          <span
            className="text-md text-primary font-semibold md:text-xl"
          >
            Already have an account ?
          </span>
           
          <a
            className="btn btn-primary btn-sm rounded-full normal-case hover:btn-secondary hover:text-base-100 md:px-8 ml-1 text-base-100"
            data-testid="loginBtn"
            href="/login"
            onClick={[Function]}
          >
            Sign In
          </a>
        </p>
      </div>
      <div
        className="hidden lg:block mb-8 lg:mb-4 text-center"
      >
        <span
          className="text-blue-600 text-center inline text-md md:text-lg"
        >
           Or sign in with 
          <a
            className="text-blue-500 ml-2"
            href="/"
            onClick={[Function]}
          >
            <svg
              className="text-2xl inline mx-1"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 8.049c0-4.446-3.582-8.05-8-8.05C3.58 0-.002 3.603-.002 8.05c0 4.017 2.926 7.347 6.75 7.951v-5.625h-2.03V8.05H6.75V6.275c0-2.017 1.195-3.131 3.022-3.131.876 0 1.791.157 1.791.157v1.98h-1.009c-.993 0-1.303.621-1.303 1.258v1.51h2.218l-.354 2.326H9.25V16c3.824-.604 6.75-3.934 6.75-7.951z"
              />
            </svg>
          </a>
          <a
            className="text-sky-400"
            href="/"
            onClick={[Function]}
          >
            <svg
              className="text-2xl inline mx-1"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.026 15c6.038 0 9.341-5.003 9.341-9.334 0-.14 0-.282-.006-.422A6.685 6.685 0 0 0 16 3.542a6.658 6.658 0 0 1-1.889.518 3.301 3.301 0 0 0 1.447-1.817 6.533 6.533 0 0 1-2.087.793A3.286 3.286 0 0 0 7.875 6.03a9.325 9.325 0 0 1-6.767-3.429 3.289 3.289 0 0 0 1.018 4.382A3.323 3.323 0 0 1 .64 6.575v.045a3.288 3.288 0 0 0 2.632 3.218 3.203 3.203 0 0 1-.865.115 3.23 3.23 0 0 1-.614-.057 3.283 3.283 0 0 0 3.067 2.277A6.588 6.588 0 0 1 .78 13.58a6.32 6.32 0 0 1-.78-.045A9.344 9.344 0 0 0 5.026 15z"
              />
            </svg>
          </a>
          <a
            className="text-red-600"
            href="/"
            onClick={[Function]}
          >
            <svg
              className="text-2xl inline mx-1"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256,8C119.1,8,8,119.1,8,256S119.1,504,256,504,504,392.9,504,256,392.9,8,256,8ZM185.3,380a124,124,0,0,1,0-248c31.3,0,60.1,11,83,32.3l-33.6,32.6c-13.2-12.9-31.3-19.1-49.4-19.1-42.9,0-77.2,35.5-77.2,78.1S142.3,334,185.3,334c32.6,0,64.9-19.1,70.1-53.3H185.3V238.1H302.2a109.2,109.2,0,0,1,1.9,20.7c0,70.8-47.5,121.2-118.8,121.2ZM415.5,273.8v35.5H380V273.8H344.5V238.3H380V202.8h35.5v35.5h35.2v35.5Z"
              />
            </svg>
          </a>
        </span>
      </div>
      <div
        className="px-5 lg:px-10 pt-5 pb-10 card mt-4 lg:mt-0 lg:mx-2 xl:mx-14 3xl:mx-64"
        role="form"
      >
        <h1
          className="card-title text-primary font-bold text-xl lg:text-2xl"
        >
          Join the Vwanu Community
        </h1>
        <div
          className="grid grid-cols-2"
        >
          <div
            className="form-control"
          >
            <label
              className="label"
            >
              <span
                className="label-text text-md text-secondary font-semibold -mb-2 mt-4"
              />
            </label>
            <input
              autoCapitalize="none"
              autoComplete="new-email"
              className="input mr-1 mt-1 lg:mt-1 bg-blue-200 text-secondary placeholder:text-secondary font-semibold rounded-full input-secondary border-none invalid:text-red-500 autofill:text-secondary autofill:bg-blue-200"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="First Name"
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="form-control"
          >
            <label
              className="label"
            >
              <span
                className="label-text text-md text-secondary font-semibold -mb-2 mt-4"
              />
            </label>
            <input
              autoCapitalize="none"
              autoComplete="new-email"
              className="input ml-1 mt-1 lg:mt-1 bg-blue-200 text-secondary placeholder:text-secondary font-semibold rounded-full input-secondary border-none invalid:text-red-500 autofill:text-secondary autofill:bg-blue-200"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Last Name"
              required={true}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="form-control"
        >
          <label
            className="label"
          >
            <span
              className="label-text text-md text-secondary font-semibold -mb-2 mt-4"
            />
          </label>
          <input
            autoCapitalize="none"
            autoComplete="new-email"
            className="input mt-1 lg:mt-1 bg-blue-200 text-secondary placeholder:text-secondary font-semibold rounded-full input-secondary border-none invalid:text-red-500 autofill:text-secondary autofill:bg-blue-200"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Email"
            required={true}
            type="email"
            value=""
          />
        </div>
        <div
          className="form-control"
        >
          <label
            className="label"
          >
            <span
              className="label-text text-md text-secondary font-semibold -mb-2 mt-4"
            />
          </label>
          <input
            autoCapitalize="none"
            autoComplete="new-email"
            autoCorrect="false"
            className="input mr-1 mt-1 lg:mt-1 bg-blue-200 text-secondary placeholder:text-secondary font-semibold rounded-full input-secondary border-none invalid:text-red-500 autofill:text-secondary autofill:bg-blue-200"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            value=""
          />
          <label
            className="w-10 rounded px-2 py-1 text-sm hover:text-gray-700 text-gray-600 text-right -mt-9 mr-3 ml-auto"
            htmlFor="toggle"
            onClick={[Function]}
          >
            Show
          </label>
        </div>
        <div
          className="form-control"
        >
          <label
            className="label"
          >
            <span
              className="label-text text-md text-secondary font-semibold -mb-2 mt-4"
            />
          </label>
          <input
            autoCapitalize="none"
            autoComplete="new-email"
            autoCorrect="false"
            className="input mr-1 mt-1 lg:mt-1 bg-blue-200 text-secondary placeholder:text-secondary font-semibold rounded-full input-secondary border-none invalid:text-red-500 autofill:text-secondary autofill:bg-blue-200"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Password Confirmation"
            required={true}
            type="password"
            value=""
          />
          <label
            className="w-10 rounded px-2 py-1 text-sm hover:text-gray-700 text-gray-600 text-right -mt-9 mr-3 ml-auto"
            htmlFor="toggle"
            onClick={[Function]}
          >
            Show
          </label>
        </div>
        <div
          className="form-control mt-3"
        >
          <label
            className="label"
          >
            <input
              className="checkbox appearance-none checked:bg-secondary checked:text-base-100 hover:text-base-100 active:text-base-100 border-blue-300 hover:border-blue-300 ml-2 lg:ml-2"
              onBlur={[Function]}
              onChange={[Function]}
              required={true}
              type="checkbox"
              value={false}
            />
            <span
              className="text-secondary font-semibold ml-2 text-sm lg:text-lg lg:ml-2"
            >
              I Agree with Privacy and Policy
            </span>
            <span
              className="ml-8 text-sm lg:text-lg"
            />
          </label>
        </div>
        <button
          className="btn btn-primary mt-4 normal-case text-base-100 hover:bg-secondary rounded-full text-md btn-md"
          data-testid="registerBtn"
          onClick={[Function]}
          onKeyDown={null}
          type="submit"
        >
          Sign Up
        </button>
      </div>
      <div
        className="lg:hidden mb-8 text-center"
      >
        <span
          className="text-blue-600 text-center inline text-md md:text-lg"
        >
           Or sign in with 
          <a
            className="text-blue-500 ml-2"
            href="/"
            onClick={[Function]}
          >
            <svg
              className="text-2xl inline mx-1"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 8.049c0-4.446-3.582-8.05-8-8.05C3.58 0-.002 3.603-.002 8.05c0 4.017 2.926 7.347 6.75 7.951v-5.625h-2.03V8.05H6.75V6.275c0-2.017 1.195-3.131 3.022-3.131.876 0 1.791.157 1.791.157v1.98h-1.009c-.993 0-1.303.621-1.303 1.258v1.51h2.218l-.354 2.326H9.25V16c3.824-.604 6.75-3.934 6.75-7.951z"
              />
            </svg>
          </a>
          <a
            className="text-sky-400"
            href="/"
            onClick={[Function]}
          >
            <svg
              className="text-2xl inline mx-1"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.026 15c6.038 0 9.341-5.003 9.341-9.334 0-.14 0-.282-.006-.422A6.685 6.685 0 0 0 16 3.542a6.658 6.658 0 0 1-1.889.518 3.301 3.301 0 0 0 1.447-1.817 6.533 6.533 0 0 1-2.087.793A3.286 3.286 0 0 0 7.875 6.03a9.325 9.325 0 0 1-6.767-3.429 3.289 3.289 0 0 0 1.018 4.382A3.323 3.323 0 0 1 .64 6.575v.045a3.288 3.288 0 0 0 2.632 3.218 3.203 3.203 0 0 1-.865.115 3.23 3.23 0 0 1-.614-.057 3.283 3.283 0 0 0 3.067 2.277A6.588 6.588 0 0 1 .78 13.58a6.32 6.32 0 0 1-.78-.045A9.344 9.344 0 0 0 5.026 15z"
              />
            </svg>
          </a>
          <a
            className="text-red-600"
            href="/"
            onClick={[Function]}
          >
            <svg
              className="text-2xl inline mx-1"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256,8C119.1,8,8,119.1,8,256S119.1,504,256,504,504,392.9,504,256,392.9,8,256,8ZM185.3,380a124,124,0,0,1,0-248c31.3,0,60.1,11,83,32.3l-33.6,32.6c-13.2-12.9-31.3-19.1-49.4-19.1-42.9,0-77.2,35.5-77.2,78.1S142.3,334,185.3,334c32.6,0,64.9-19.1,70.1-53.3H185.3V238.1H302.2a109.2,109.2,0,0,1,1.9,20.7c0,70.8-47.5,121.2-118.8,121.2ZM415.5,273.8v35.5H380V273.8H344.5V238.3H380V202.8h35.5v35.5h35.2v35.5Z"
              />
            </svg>
          </a>
        </span>
      </div>
    </div>
  </div>
</div>
`;
